README.md

```diff
@@ -38,765 +56,196 @@ Given that Webpacker already provides React integration, why would you use "Reac
+### React Router
 
-#### Redux Store Notes
-Note: you don't need to initialize your redux store. You can pass the props to your React component in a "generator function." However, consider using the `redux_store` helper for the two following use cases:
-
-1. You want to have multiple React components accessing the same store at once.
-2. You want to place the props to hydrate the client side stores at the very end of your HTML so that the browser can render all earlier HTML first. This is particularly useful if your props will be large.
-
-### server_render_js
-`server_render_js(js_expression, options = {})`
-
-+ js_expression, like 2 + 3, and not a block of js code. If you have more than one line that needs to be executed, wrap it in an [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression). JS exceptions will be caught, and console messages will be handled properly
-+ Currently, the only option you may pass is `replay_console` (boolean)
-
-This is a helper method that takes any JavaScript expression and returns the output from evaluating it. If you have more than one line that needs to be executed, wrap it in an IIFE. JS exceptions will be caught and console messages handled properly.
+[React Router](https://github.com/reactjs/react-router) is supported, including server-side rendering! See:
 
-## Multiple React Components on a Page with One Store
-You may wish to have 2 React components share the same the Redux store. For example, if your navbar is a React component, you may want it to use the same store as your component in the main area of the page. You may even want multiple React components in the main area, which allows for greater modularity. Also, you may want this to work with Turbolinks to minimize reloading the JavaScript. A good example of this would be something like a notifications counter in a header. As each notification is read in the body of the page, you would like to update the header. If both the header and body share the same Redux store, then this is trivial. Otherwise, we have to rely on other solutions, such as the header polling the server to see how many unread notifications exist.
+1. [React on Rails docs for react-router](docs/additional-reading/react-router.md)
+2. Examples in [spec/dummy/app/views/react_router](./spec/dummy/app/views/react_router) and follow to the JavaScript code in the [spec/dummy/client/app/startup/ServerRouterApp.jsx](spec/dummy/client/app/startup/ServerRouterApp.jsx).
+3. [Code Splitting docs](docs/additional-reading/code-splitting.md) for information about how to set up code splitting for server rendered routes.
 
-Suppose the Redux store is called `appStore`, and you have 3 React components that each needs to connect to a store: `NavbarApp`, `CommentsApp`, and `BlogsApp`. I named them with `App` to indicate that they are the registered components.
+# Caching and Performance
```

Comment:
We should:
- go for a `##` instead of a `#` which [should only be used for the document title](https://github.com/markdownlint/markdownlint/blob/e3dbc768e1608da5baab0a5896c54e840ff59110/docs/RULES.md#md025---multiple-top-level-headers-in-the-same-document) 
- add a [blank line before the header](https://github.com/markdownlint/markdownlint/blob/e3dbc768e1608da5baab0a5896c54e840ff59110/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines).

By doing so we would be incrementing header levels by one level at a time as [suggested](https://github.com/markdownlint/markdownlint/blob/e3dbc768e1608da5baab0a5896c54e840ff59110/docs/RULES.md#md001---header-levels-should-only-increment-by-one-level-at-a-time).